### PR TITLE
uniform payment messages. introduce new super class PaymentMessage

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-766

### Change description ###
introduce  super class PaymentMessege. in next PRs will try to use same PaymentMessageProcessor
for all payment messages

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java [hunk 1]
```diff
@@ -76,11 +77,11 @@ public boolean processNextMessage() throws ServiceBusException, InterruptedExcep
     private MessageProcessingResult processCreateCommand(IMessage message) {
         log.info("Started processing payment message with ID {}", message.getMessageId());
 
-        CreatePaymentMessage payment = null;
+        PaymentMessage payment = null;
 
         try {
             payment = paymentMessageParser.parse(message.getMessageBody());
-            paymentMessageHandler.handlePaymentMessage(payment);
+            paymentMessageHandler.handlePaymentMessage((CreatePaymentMessage)payment);
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java]: Casting to CreatePaymentMessage is bad

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java]: next PR.

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java]: perhaps should apply interface across the codebase and only cast where it's needed ;). or some sort of way because `handlePaymentMesage` returns specific response only related to create event

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java]: > perhaps should apply interface across the codebase and only cast where it's needed ;). or some sort of way because `handlePaymentMesage` returns specific response only related to create event

at the end  planning tot do sth like this #72

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java [hunk 2]
```diff
@@ -76,11 +77,11 @@ public boolean processNextMessage() throws ServiceBusException, InterruptedExcep
     private MessageProcessingResult processCreateCommand(IMessage message) {
         log.info("Started processing payment message with ID {}", message.getMessageId());
 
-        CreatePaymentMessage payment = null;
+        PaymentMessage payment = null;
 
         try {
             payment = paymentMessageParser.parse(message.getMessageBody());
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java]: this call returns `CreatePaymentMessage`. why this change??

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java]: PR #72  would give an idea what  I am trying to do.
I do not want to change Processor for every new payment process.

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java]: completely different things happen based on the type of payment message, so why not?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java]: anyway, this change doesn't make sense in the scope of this PR of adding super class

**Comment by mdayican** [comment]
want to continue with #82
